File: modules/structured_output/exercises/01_extract_json/solution/cpp/main.cpp
#include <nlohmann/json.hpp>
#include <iostream>
#include <string>

int main() {
    std::string raw = R"(Result: {"label":"positive"})";
    size_t start = raw.find('{');
    size_t end = raw.rfind('}');
    auto data = nlohmann::json::parse(raw.substr(start, end - start + 1));
    std::cout << "EXTRACT_OK:" << data.at("label").get<std::string>() << "\n";
    return 0;
}

File: modules/structured_output/exercises/01_extract_json/solution/python/main.py
def main():
    raw = 'Result: {"label":"positive"}'
    start, end = raw.find("{"), raw.rfind("}")+1
    import json
    data = json.loads(raw[start:end])
    print(f"EXTRACT_OK:{data['label']}")

if __name__ == "__main__": main()
